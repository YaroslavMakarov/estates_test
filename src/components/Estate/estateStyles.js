import styled from "styled-components";

export const EstateContainer = styled.div`
    box-sizing: border-box;
    padding: 15px;
`;
export const Img = styled.img.attrs(props => ({
    src: props.src
}))`
    display: block;
    width: 100%;
    height: 300px;
    object-fit: cover;
    margin-bottom: 10px;
`;
export const EstateDescription = styled.div`
    font-size: 16px;
    margin-bottom: 5px;
`;