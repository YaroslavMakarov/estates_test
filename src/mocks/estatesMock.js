export const estatesMock = {
    estates: {
        estates: [
            {
                'id': 1,
                'title': 'Shevchenko estate',
                'price': 30000,
                'address': {
                    'street': 'Melnychna',
                    'number': 12
                },
                'preview_img': 'https://cp12.nevsepic.com.ua/62/1353962605-0435309-www.nevsepic.com.ua.jpg',
                'seller': 'Chack Norris',
                'description': 'The best from Chuck',
                'detailed_photos': [
                    'https://www.wtoc.com/resizer/ckPwhlUdYaFcLiBWMX0J033a0Ow=/1200x0/arc-anglerfish-arc2-prod-raycom.s3.amazonaws.com/public/JGJB35IW4JAMROXX3D3MMEF2SU.jpg',
                    'https://img3.goodfon.ru/wallpaper/nbig/c/f7/chuck-norris-chak-norris-legenda-aktior-vzgliad.jpg',
                    'https://cdn.flickeringmyth.com/wp-content/uploads/2020/03/chuck-norris.jpg'
                ]
            },
            {
                "id": 2,
                "title": "Yanukovich estate",
                "price": 3000000,
                "address": {
                    "street": "Prisident",
                    "number": 4
                },
                "preview_img": "https://biz.liga.net/images/general/2018/11/15/thumbnail-tw-20181115111659-8474.jpg?v=1542279867",
                "seller": "Chack Norris",
                "description": "The best from Chuck",
                "detailed_photos": [
                    "https://www.wtoc.com/resizer/ckPwhlUdYaFcLiBWMX0J033a0Ow=/1200x0/arc-anglerfish-arc2-prod-raycom.s3.amazonaws.com/public/JGJB35IW4JAMROXX3D3MMEF2SU.jpg",
                    "https://img3.goodfon.ru/wallpaper/nbig/c/f7/chuck-norris-chak-norris-legenda-aktior-vzgliad.jpg",
                    "https://cdn.flickeringmyth.com/wp-content/uploads/2020/03/chuck-norris.jpg"
                    ]
             },
        ],
    },
};