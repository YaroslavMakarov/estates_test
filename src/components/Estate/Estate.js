import React from "react";
import { EstateContainer, Img, EstateDescription } from "./estateStyles"

const Estate = ({ estate }) => {
    return (
        <EstateContainer>
            <Img src={estate.preview_img} />
            <EstateDescription>
                {`Name: ${estate.title}`}
            </EstateDescription>
            <EstateDescription>
                {`Price: ${estate.price}$`}
            </EstateDescription>
            <EstateDescription>
                {`Address: ${estate.address.street}, ${estate.address.number}`}
            </EstateDescription>
        </EstateContainer>
    )
};

export default Estate;
