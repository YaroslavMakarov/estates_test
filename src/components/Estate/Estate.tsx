import React from 'react';
import { Link } from 'react-router-dom';

import { EstateContainer, Img, EstateDescription } from '../../styled/estateStyles';

type Props = {
    estate: Estate;
};

const Estate: React.FC<Props> = ({ estate }) => {

    return (
        <EstateContainer
          as={Link}
          to={`/info/${estate.id}`}
        >
            <Img src={estate.preview_img} />
            <EstateDescription>
                {`Name: ${estate.title}`}
            </EstateDescription>
            <EstateDescription>
                {`Price: ${estate.price}$`}
            </EstateDescription>
            <EstateDescription>
                {`Address: ${estate.address.street}, ${estate.address.number}`}
            </EstateDescription>
        </EstateContainer>
    )
};

export default Estate;
