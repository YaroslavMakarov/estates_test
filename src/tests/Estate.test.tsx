import React from 'react';
import { render, RenderResult, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import Estate from '../components/Estate';
import { estatesMock } from '../mocks/estatesMock';

describe('Estate component', () => {
    let wrapper: RenderResult;
    const estate: Estate = estatesMock.estates.estates[0];

    beforeEach(() => {
        wrapper = render(    
            <MemoryRouter>
                <Estate estate={estate} />
            </MemoryRouter>    
        );
    });

    it('get snapshot Estate', () => {
        expect(wrapper.baseElement).toMatchSnapshot();
    });

    it('check Link attr', () => {
        expect(screen.getByTestId('estate')).toHaveAttribute('href', `/info/${estate.id}`);
    });

    it('check Img attr', () => {
        expect(screen.getByAltText('main img')).toHaveAttribute('src', `${estate.preview_img}`);
    });

    it('check Estate name', () => {
        expect(screen.getByText(`Name: ${estate.title}`)).toBeTruthy();
    });

    it('check Estate pricce', () => {
        expect(screen.getByText(`Price: ${estate.price}$`)).toBeTruthy();
    });

    it('check Estate address', () => {
        expect(screen.getByText(`Address: ${estate.address.street}, ${estate.address.number}`)).toBeTruthy();
    });
});