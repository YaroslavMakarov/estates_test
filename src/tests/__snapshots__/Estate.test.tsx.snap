// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Estate component get snapshot Estate 1`] = `
<body>
  <div>
    <a
      class="sc-bdfBwQ fhkZDT"
      data-testid="estate"
      href="/info/1"
    >
      <img
        alt="main img"
        class="sc-gsTCUz esHetz"
        src="https://cp12.nevsepic.com.ua/62/1353962605-0435309-www.nevsepic.com.ua.jpg"
      />
      <div
        class="sc-dlfnbm hMEdSQ"
      >
        Name: Shevchenko estate
      </div>
      <div
        class="sc-dlfnbm hMEdSQ"
      >
        Price: 30000$
      </div>
      <div
        class="sc-dlfnbm hMEdSQ"
      >
        Address: Melnychna, 12
      </div>
    </a>
  </div>
</body>
`;
